src/runtime/server/middleware/session/index.ts

```diff
@@ -121,6 +121,19 @@ const getSession = async (event: H3Event): Promise<null | Session> => {
   return session
 }
 
+const getImmutableSession = (session: Session) => {
+  const immutableSession = { ...session }
+  const properties = ['id', 'createdAt', 'ip']
+
+  properties.forEach((property) => {
+    Object.defineProperty(immutableSession, property, {
+      writable: false
```

Comment:
let's also add `configurable: false` so that this property cannot be changed back to writable: https://stackoverflow.com/a/42853662